src/coord/src/coord.rs

```diff
@@ -1356,7 +1344,7 @@ where
                 typ: _,
             } = source.as_ref()
             {
-                if let Some(Some((index_id, _))) = self.views.get(&id).map(|v| &v.default_idx) {
+                if let Some((index_id, _)) = self.catalog.indexes()[&id].first() {
```

Comment:
Why `first()` here?

Reply:
That's just the convention for the "default" index. But you're right that that's very unclear. I added a `Catalog::default_index_for` method that makes things read more nicely.